src/Workspaces/Core/Portable/TemporaryStorage/TemporaryStorageServiceFactory.cs

```diff
@@ -487,7 +493,36 @@ public override int Read(char[] buffer, int index, int count)
 
                 return count;
             }
+
+            public override int ReadBlock(char[] buffer, int index, int count) =>
+                Read(buffer, index, count);
+
+#if NETCOREAPP
+            public override int Read(Span<char> buffer)
+            {
+                var count = Math.Min(buffer.Length, (int)(_end - _position));
+                if (count > 0)
+                {
+                    new ReadOnlySpan<char>(_position, count).CopyTo(buffer);
+                    _position += count;
+                }
+
+                return count;
+            }
+
+            public override int ReadBlock(Span<char> buffer) =>
+                Read(buffer);
+#endif
+
+            public override string ReadToEnd()
+            {
+                if (_position >= _end)
+                    return "";
```

Comment:
interesting.  is the `> end` case expected?  or can you only have the `==` case?

Reply:
It shouldn't be needed if everything is correct. I just used to using it likes this defensively. Also, this matches the `count` checks in the other methods where it checks that `count` is greater than zero, as opposed to not equals zero. I can changed this to `==` if you feel that is more appropriate.

Reply:
That would be great.  Can you switch to that, and put a contract call in that _position must be `<=` to _end`?  Thanks!

Reply:
Fixed.